RangeError [ERR_OUT_OF_RANGE]

**Version:**
`native-run: 0.2.7`
`node: 10.16.0`

<!-- Please post the output of "native-run --version" -->


**Description:**
Native-run fails when running both through `ionic run ios` and when running `native-run` directly. I've tried running `ionic run ios` both with and without the `--l` flag with same result. Tried running `native-run ios --app "platforms/ios/build/device/App Admin.ipa" --device --connect --forward 8100:8100` with same result.

<!-- Please describe the problem you're having. -->


**Command Output:**
```
RangeError [ERR_OUT_OF_RANGE]: The value of "offset" is out of range. It must be >= 0 and <= 37. Received 38
    at boundsError (internal/buffer.js:49:9)
    at writeU_Int8 (internal/buffer.js:558:5)
    at Buffer.writeUInt8 (internal/buffer.js:565:10)
    at toCString (/Users/antongildebrand/.nvm/versions/node/v10.16.0/lib/node_modules/native-run/node_modules/node-ioslib/dist/client/afc.js:150:9)
    at AFCClient.makeDirectory (/Users/antongildebrand/.nvm/versions/node/v10.16.0/lib/node_modules/native-run/node_modules/node-ioslib/dist/client/afc.js:84:19)
    at AFCClient.uploadDirectory (/Users/antongildebrand/.nvm/versions/node/v10.16.0/lib/node_modules/native-run/node_modules/node-ioslib/dist/client/afc.js:89:20)
    at uploadApp (/Users/antongildebrand/.nvm/versions/node/v10.16.0/lib/node_modules/native-run/dist/ios/utils/device.js:85:21)
    at process._tickCallback (internal/process/next_tick.js:68:7)
```
<!-- IMPORTANT: Please run your command with the "--verbose" flag to show all debugging output. -->


**Environment Info:**
```
Ionic:

   Ionic CLI                     : 5.2.1 (/Users/antongildebrand/.nvm/versions/node/v10.16.0/lib/node_modules/ionic)
   Ionic Framework               : @ionic/angular 4.5.0
   @angular-devkit/build-angular : 0.13.9
   @angular-devkit/schematics    : 7.3.9
   @angular/cli                  : 7.3.9
   @ionic/angular-toolkit        : 1.5.1

Cordova:

   Cordova CLI       : 9.0.0 (cordova-lib@9.0.1)
   Cordova Platforms : ios 5.0.1
   Cordova Plugins   : cordova-plugin-ionic-keyboard 2.1.3, cordova-plugin-ionic-webview 4.1.1, (and 8 other plugins)

Utility:

   cordova-res : not installed
   native-run  : 0.2.7

System:

   Android SDK Tools : 26.1.1 (/Users/antongildebrand/Library/android-sdk-macosx)
   ios-deploy        : 1.9.4
   ios-sim           : 8.0.1
   NodeJS            : v10.16.0 (/Users/antongildebrand/.nvm/versions/node/v10.16.0/bin/node)
   npm               : 6.9.0
   OS                : macOS Mojave
   Xcode             : Xcode 10.2.1 Build version 10E1001
```
<!-- If you're using Ionic, you can simply post the output of "ionic info" -->

macOS version:
`10.14.5`

Xcode version:
`10.2.1`

I got a similar error when I run "ionic cordova run ios -l --external --device" 
All works good if I run it from xCode or the browser though. 

The output :

```
> native-run ios --app platforms/ios/build/device/Preprod.ipa --device --forward 8100:8100
[native-run] RangeError [ERR_OUT_OF_RANGE]: The value of "offset" is out of range. It must be >= 0 and <= 85. Received 86
[native-run]     at boundsError (internal/buffer.js:77:9)
[native-run]     at writeU_Int8 (internal/buffer.js:732:5)
[native-run]     at Buffer.writeUInt8 (internal/buffer.js:739:10)
[native-run]     at toCString (/usr/local/lib/node_modules/native-run/node_modules/node-ioslib/dist/client/afc.js:150:9)
[native-run]     at AFCClient.openFile (/usr/local/lib/node_modules/native-run/node_modules/node-ioslib/dist/client/afc.js:50:9)
[native-run]     at AFCClient.uploadFile (/usr/local/lib/node_modules/native-run/node_modules/node-ioslib/dist/client/afc.js:69:24)
[ERROR] An error occurred while running subprocess native-run.
```

--- a/src/ios/lib/client/afc.ts
+++ b/src/ios/lib/client/afc.ts
@@ -52,7 +52,7 @@ export class AFCClient extends ServiceClient<AFCProtocolClient> {
   async openFile(path: string): Promise<Buffer> {
     debug(`openFile: ${path}`);
     // mode + path + null terminator
-    const data = Buffer.alloc(8 + path.length + 1);
+    const data = Buffer.alloc(8 + Buffer.byteLength(path) + 1);
     // write mode
     data.writeUInt32LE(AFC_FILE_OPEN_FLAGS.WRONLY, 0);
     // then path to file
@@ -176,7 +176,7 @@ export class AFCClient extends ServiceClient<AFCProtocolClient> {
 }
 
 function toCString(s: string) {
-  const buf = Buffer.alloc(s.length + 1);
+  const buf = Buffer.alloc(Buffer.byteLength(s) + 1);
   const len = buf.write(s);
   buf.writeUInt8(0, len);
   return buf;

